jmh-core/src/main/java/org/openjdk/jmh/runner/Runner.java

```diff
@@ -181,6 +181,9 @@ public RunResult runSingle() throws RunnerException {
      * @throws org.openjdk.jmh.runner.RunnerException if something goes wrong
      */
     public Collection<RunResult> run() throws RunnerException {
+        if (JMH_LOCK_IGNORE) {
+            return internalRun();
```

Comment:
We should still print the warning here. `-Djmh.ignoreLock=true` must not go unnoticed, because it has a potential to break performance benchmarks. (I guess it is a separate discussion why does any plugin pass it in -- it should not without user knowing!). So, I suggest:

```
 if (JMH_LOCK_IGNORE) {
    out.println("# WARNING: JMH lock is ignored by user request, make sure no other JMH instances are running");
    return internalRun();
 }
```